### PR TITLE
User and Relationship models

### PR DESCRIPTION
**Pull request by Vigrouze**
Hey guys,
We created the User and Relationship models with Alex ;) Could you please check our code ?
Thx

### CODE REVIEW

**Review by Joelebucheron — COMMENTED** [review]
see Nico

File: db/migrate/20211122155214_add_columns_to_users.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class AddColumnsToUsers < ActiveRecord::Migration[6.0]
+  def change
+    add_column :users, :first_name, :string
+    add_column :users, :last_name, :string
+    add_column :users, :address, :string
+    add_column :users, :language, :string
+    add_column :users, :description, :string
+    add_column :users, :link, :string
```
Comment (ndgdl) [inline db/migrate/20211122155214_add_columns_to_users.rb:8]: As discussed, need to add more specific link:
add_column :users, :link_github, :string
add_column :users, :link_malt, :string
add_column :users, :link_linkedin, :string
add_column :users, :link_slack, :string

**Review by Joelebucheron — APPROVED** [review]

File: app/models/relationship.rb [hunk 2]
```diff
@@ -0,0 +1,3 @@
+class Relationship < ApplicationRecord
+  belongs_to :user
```
Comment (ndgdl) [inline app/models/relationship.rb]: Need to replace 
`belongs_to :user`

by:
```
 belongs_to :mentor, class_name: "User"
 belongs_to :padawan, class_name: "User"
```

File: app/models/user.rb [hunk 3]
```diff
@@ -1,6 +1,9 @@
 class User < ApplicationRecord
   # Include default devise modules. Others available are:
   # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  belongs_to :padawan, class_name: "User"
+  belongs_to :mentor, class_name: "User"
+
```
Comment (ndgdl) [inline app/models/user.rb]: Needs to be replaced with:

```
has_many :relationships_as_mentor, class_name: "Relationship", foreign_key: :mentor_id
has_many :relationships_as_padawan, class_name: "Relationship", foreign_key: :padawan_id
```

File: db/migrate/20211122160739_create_relationships.rb [hunk 4]
```diff
@@ -0,0 +1,11 @@
+class CreateRelationships < ActiveRecord::Migration[6.0]
+  def change
+    create_table :relationships do |t|
+      t.references :padawan, null: false, foreign_key: { to_table: :users }
+      t.references :mentor, null: false, foreign_key: { to_table: :users }
+      t.integer :statut
```
Comment (ndgdl) [inline db/migrate/20211122160739_create_relationships.rb:6]: Change to "status" for English? 😬

**Review by ndgdl — CHANGES_REQUESTED** [review]
A new migration to do for Users?
And small adjustments to the Relationship and User model!

**Review by ndgdl — APPROVED** [review]

**Pull request merged**